### PR TITLE
Split testing-jvm subproject into testing-jvm-infrastructure

### PR DESCRIPTION
**Pull request by jvandort**
Splits `testing-jvm` into a new subproject, `testing-jvm-infrastructure`. This new subproject is specifically intended for classes which are instantiated on test workers, and thus have a smaller set of dependencies and a lower target JVM version than the existing `testing-jvm` project.

Making this distinction allows us to better recognize and control the dependencies which are required for running tests on test workers. Further, this allows us to remove `usedInWorkers` for `testing-jvm`, thus allowing us to compile non-infrastructure test-related classes with newer versions of java.

Future work includes moving test-related classes from the `plugins` subproject to the `testing-jvm` project, now that the `testing-jvm` project is no longer restricted to compiling against java 6.

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test qfl

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

**Review by tresat — APPROVED** [review]

File: subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformSpec.java [hunk 1]
```diff
@@ -28,13 +27,20 @@ public class JUnitPlatformSpec extends AbstractJUnitSpec {
     private final Set<String> includeTags;
     private final Set<String> excludeTags;
 
-    public JUnitPlatformSpec(JUnitPlatformOptions options, Set<String> includedTests,
-                             Set<String> excludedTests, Set<String> includedTestsCommandLine) {
+    public JUnitPlatformSpec(
```
[the remark this answers is not in the document]
Comment (jvandort) [reply subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformSpec.java:30]: Yup -- gonna follow this up with another PR I already wrote, which I've got stashed away somewhere now...

File: subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/IgnoredTestDescriptorProvider.java [hunk 2]
```diff
@@ -71,15 +67,15 @@ static Runner getRunnerLegacy(Class<?> testClass) throws Throwable {
                 return runnerClass.getConstructor(Class.class).newInstance(testClass);
             } catch (NoSuchMethodException e) {
                 String simpleName = runnerClass.getSimpleName();
-                throw new InitializationError("Custom runner class " + simpleName +
+                throw new org.junit.internal.runners.InitializationError("Custom runner class " + simpleName +
```
[the remark this answers is not in the document]
Comment (tresat) [reply subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/IgnoredTestDescriptorProvider.java:70]: Got it.  I didn't notice the deprecation suppression on the method.

**Pull request merged**